Docker client panics when deploying compose file

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#reporting-other-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- Docker Support Forums - https://forums.docker.com
- IRC - irc.freenode.net #docker channel
- Post a question on StackOverflow, using the Docker tag

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

When I deploy a file which contains a service connected to an external network, and I try
to provide an alias for the container on said connection, the docker client panics with a
segmentation fault.

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

1. Initialize a swarm
2. Deploy a compose file with network connection as described
3. Get a SIGSEGV!

**Describe the results you received:**

The following output:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x66e588]

goroutine 1 [running]:
panic(0xab4480, 0xc420012080)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/docker/docker/cli/command/stack.convertServiceNetworks(0xc4201a8420, 0x7ffe3ee2a82a, 0xd, 0xc4203ec118, 0x6, 0x0, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/cli/command/stack/deploy.go:255 +0x168
github.com/docker/docker/cli/command/stack.convertService(0x7ffe3ee2a82a, 0xd, 0xc4203ec118, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/src/docker/.gopath/src/github.com/docker/docker/cli/command/stack/deploy.go:519 +0x42e
github.com/docker/docker/cli/command/stack.convertServices(0x7ffe3ee2a82a, 0xd, 0xc42031b4d0, 0x7ffe3ee2a82a, 0xd, 0xc4201a8720)
        /usr/src/docker/.gopath/src/github.com/docker/docker/cli/command/stack/deploy.go:446 +0x15b
github.com/docker/docker/cli/command/stack.deployCompose(0xc4201d3440, 0x0, 0x0, 0x7ffe3ee2a817, 0x12, 0x7ffe3ee2a82a, 0xd, 0x0, 0x4, 0xc4203ec018)
        /usr/src/docker/.gopath/src/github.com/docker/docker/cli/command/stack/deploy.go:108 +0x572
github.com/docker/docker/cli/command/stack.runDeploy(0xc4201d3440, 0x0, 0x0, 0x7ffe3ee2a817, 0x12, 0x7ffe3ee2a82a, 0xd, 0x0, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/cli/command/stack/deploy.go:69 +0xe5
github.com/docker/docker/cli/command/stack.newDeployCommand.func1(0xc4203c98c0, 0xc4203c2e40, 0x1, 0x3, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/cli/command/stack/deploy.go:48 +0x93
github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).execute(0xc4203c98c0, 0xc42000c0c0, 0x3, 0x3, 0xc4203c98c0, 0xc42000c0c0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:646 +0x26d
github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200918c0, 0xc420091d40, 0xc420348f00, 0xc42002c440)
        /usr/src/docker/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:742 +0x377
github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200918c0, 0xc4200918c0, 0x10e7ce0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:695 +0x2b
main.main()
        /usr/src/docker/.gopath/src/github.com/docker/docker/cmd/docker/docker.go:93 +0xcb
```

**Describe the results you expected:**

A successfully deployed compose file

**Additional information you deem important (e.g. issue happens only occasionally):**

Happens consistently

**Output of `docker version`:**

```
Client:
 Version:      1.13.0-rc1
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   75fd88b
 Built:        Fri Nov 11 19:47:07 2016
 OS/Arch:      linux/amd64

Server:
 Version:             1.13.0-rc1
 API version:         1.25
 Minimum API version: 1.12
 Go version:          go1.7.3
 Git commit:          75fd88b
 Built:               Fri Nov 11 19:47:07 2016
 OS/Arch:             linux/amd64
 Experimental:        true
```

**Output of `docker info`:**

```
Containers: 129
 Running: 4
 Paused: 0
 Stopped: 125
Images: 296
Server Version: 1.13.0-rc1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 912
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: 8agsrahzgub7m48dr66508iuv
 Is Manager: true
 ClusterID: 9l3fu4qrystn2d4h5dgly9uhc
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.2.52
 Manager Addresses:
  192.168.2.52:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 8517738ba4b82aff5662c97ca4627e7e4d03b531
runc version: ac031b5bf1cc92239461125f4c1ffb760522bbf2
init version: N/A (expected: v0.13.0)
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-45-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.56 GiB
Name: docker-x86-1
ID: NKCQ:BOL4:6JUP:BUHD:LU46:KLAQ:FDKZ:VP7I:U5FN:VYDI:OS2J:SP3K
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Labels:
 provider=generic
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical machine running Ubuntu 16.04 LTS

--- a/cli/command/stack/deploy.go
+++ b/cli/command/stack/deploy.go
@@ -252,9 +252,13 @@ func convertServiceNetworks(
 
 	nets := []swarm.NetworkAttachmentConfig{}
 	for networkName, network := range networks {
+		var aliases []string
+		if network != nil {
+			aliases = network.Aliases
+		}
 		nets = append(nets, swarm.NetworkAttachmentConfig{
 			Target:  namespace.scope(networkName),
-			Aliases: append(network.Aliases, name),
+			Aliases: append(aliases, name),
 		})
 	}
 	return nets

